### PR TITLE
Tuples: PR feedback on #10607

### PR DESCRIPTION
**Pull request by jcouv**
@AlekseyTs for review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs [hunk 1]
```diff
@@ -387,26 +387,32 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
 
         private void AddTupleTypeName(INamedTypeSymbol symbol)
         {
-            var tupleType = (TupleTypeSymbol)symbol;
+            var tupleType = symbol as TupleTypeSymbol;
+            if (tupleType == null)
+            {
+                return;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs]: If we simply return, nothing will be added. I would expect us to at least fallback to the underlying type. Can we get one from INamedTypeSymbol? If not, consider adding an API next to IsTupleType.

**Comment by AlekseyTs** [comment]
LGTM, modulo the issue above.

**Comment by VSadov** [comment]
LGTM

**Comment by jcouv** [comment]
With the latest commit, `SymbolDisplayVisitor` can display any type that returns `IsTupleType` as true, and without assuming it must be a `TupleTypeSymbol`. 
A test is added to cover this specific case with a mock symbol.

**Comment by jcouv** [comment]
@AlekseyTs The fix above is as we discussed. I wasn't able to re-use a mock for `NamedTypeSymbol`, but I made one with only a few methods actually implemented, for testing.
Let me know if I can go ahead with merge. Thanks

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**